### PR TITLE
feat: add real_ip_recursive to nginx.conf

### PR DESCRIPTION
**Pull request by keilon**
### What this PR does / why we need it:
Enable to set real_ip_recursive to nginx.conf. When apisix is behind multiple layer network, the real client ip may not be recognized. See also [http://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_recursive](http://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_recursive)

### Pre-submission checklist:

* [X] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [X] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by Firstsawyou — COMMENTED** [review]
We should add corresponding test cases.  https://github.com/apache/apisix/blob/master/t/cli/test_main.sh

File: apisix/cli/ngx_tpl.lua [hunk 1]
```diff
@@ -229,6 +229,10 @@ http {
     real_ip_header {* http.real_ip_header *};
     {% end %}
 
+    {% if http.real_ip_recursive then %}
+    real_ip_recursive {* http.real_ip_recursive *};
```
Comment (spacewander) [inline apisix/cli/ngx_tpl.lua:233]: Need to add a test like: https://github.com/apache/apisix/blob/9a40fd7cb497974f9d7bd75f128eed5dd238cdc6/t/cli/test_main.sh#L382-L399

File: t/cli/test_main.sh [hunk 2]
```diff
@@ -621,3 +621,10 @@ if ! grep "keepalive_timeout 6s;" conf/nginx.conf > /dev/null; then
 fi
 
 echo "passed: found the keepalive related parameter in nginx.conf"
+
+if ! grep "real_ip_recursive off;" conf/nginx.conf > /dev/null; then
+    echo "failed: 'real_ip_recursive off;' not in nginx.conf"
+    exit 1
+fi
+
+echo "passed: disable real_ip_recursive by default in nginx.conf"
```
Comment (spacewander) [inline t/cli/test_main.sh:630]: Need a test to change its default value and check, like `client_max_body_size: 512m ` in my example.

Comment (keilon) [reply t/cli/test_main.sh]: Fixed.

**Review by tokers — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**